### ISSUE TITLE
Loading a deployment from server with a config file throws exception

### ISSUE THREAD

**Issue by kkucera** [fabric8io/kubernetes-client#807]
The following call with a config file of kind: Deployment causes an exception:

```
KubernetesClient client = new DefaultKubernetesClient();
List<HasMetadata> deploymentList = client.load(getFileInputString("deployment.yml")).inNamespace("mynamespace").fromServer().get();
```

```
Exception in thread "main" java.lang.ClassCastException: io.fabric8.kubernetes.api.model.extensions.DeploymentBuilder cannot be cast to io.fabric8.kubernetes.client.HasMetadataVisitiableBuilder
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.get(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.java:270)
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.get(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.java:66)
```
I'm using kubernetes-client: 2.5.2 which pulls kubernetes-model: 1.0.78

I'm not sure if this is an issue in the kubernetes-client project or the kubernetes-model project.
Seems like `io.fabric8.kubernetes.api.model.extensions.DeploymentBuilder` is supposed to be implementing `io.fabric8.kubernetes.client.HasMetadataVisitiableBuilder`

**Comment by uce** [comment]
This issue still persists for `kubernetes-client:3.1.11` with `kubernetes-model:2.0.9`. For a similar operation as shown by @kkucera for a `ConfigMap`, I get:

```
java.lang.ClassCastException: io.fabric8.kubernetes.api.model.ConfigMapBuilder cannot be cast to io.fabric8.kubernetes.client.HasMetadataVisitiableBuilder
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.get(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java:173)
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.get(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableImpl.java:59)
```

**Comment by rohanKanojia** [comment]
closed via #1434
